pkg/operator/k8sutil/configmap.go

```diff
@@ -62,13 +64,30 @@ func GetOperatorSetting(context context.Context, clientset kubernetes.Interface,
 }
 
 func GetValue(data map[string]string, settingName, defaultValue string) string {
-	if settingValue, ok := data[settingName]; ok {
-		logger.Infof("%s=%q (configmap)", settingName, settingValue)
-		return settingValue
-	} else if settingValue, ok := os.LookupEnv(settingName); ok {
-		logger.Infof("%s=%q (env var)", settingName, settingValue)
-		return settingValue
+	var settingValue, settingSource string
```

Comment:
```suggestion
var settingValue=defaultValue
var settingSource = "default"
if val, ok := data[settingName]; ok {
	settingValue = val
	settingSource = "configmap"
} else if val, ok := os.LookupEnv(settingName); ok {
	settingValue = val
	settingSource = "env var"
}
```

If we set the var as defaults you can avoid one else block?

Also, this configmap will be accessed by multiple controllers at the same time, I think we may have a race, you might need to use sync.Map, please check

Reply:
Agreed, good catch to make sure it's safe across the controllers.